libtiledbsoma/src/reindexer/reindexer.cc

```diff
@@ -46,28 +46,37 @@ KHASH_MAP_INIT_INT64(m64, int64_t)
 
 namespace tiledbsoma {
 
-void IntIndexer::map_locations(
-    const int64_t* keys, size_t size, size_t threads) {
+std::shared_ptr<tiledbsoma::ThreadPool>
+    IntIndexer::tiledb_thread_pool_ = nullptr;
+
+void IntIndexer::start_thread_pool(int thread_count) {
+    tiledb_thread_pool_ = std::make_unique<tiledbsoma::ThreadPool>(
+        thread_count);
+    LOG_DEBUG(fmt::format(
+        "[Re-indexer] Thread pool started with {} threads", thread_count));
+}
+
+void IntIndexer::stop_thread_pool() {
+    tiledb_thread_pool_.reset();
+    LOG_DEBUG("[Re-indexer] Thread stopped");
+}
```

Comment:
Apart from the problems noted above, the way this thread pool is accessed globally would need lock protection—otherwise, the same pointer can be racily accessed and reset across multiple threads with no guarantee of atomicity, leading to the dreaded Undefined Behavior.

Reply:
I'll work on implementing the design I mentioned @thetorpedodog @bkmartinjr